src/main/java/com/android/volley/CacheDispatcher.java

```diff
@@ -67,12 +63,13 @@ public CacheDispatcher(
             BlockingQueue<Request<?>> cacheQueue,
             BlockingQueue<Request<?>> networkQueue,
             Cache cache,
-            ResponseDelivery delivery) {
+            ResponseDelivery delivery,
+            WaitingRequestManager waitingRequestManager) {
         mCacheQueue = cacheQueue;
         mNetworkQueue = networkQueue;
```

Comment:
This changes the public API and will break for anyone who implements their own RequestQueue. It's probably better to also keep the current constructor, maybe mark it as deprecated.

Reply:
CacheDispatcher is public. I don't think you can assume that removing a public constructor will be backwards compatible, regardless of what the expected use is.